reports/tasks.py

```diff
@@ -0,0 +1,226 @@
+        sys.stdout.write(line)
+    log_line.info = log_line
+    log_line.error = log_line
+
+    playbook_extra_script_arguments = '--skip-hit-statistics' if settings.INSTANCE_LOGS_SERVER_HOST else ''
+
+    # Launch the collect_activity playbook, which places a file into the `playbook_output_dir`
+    # on this host.
+    ansible.capture_playbook_output(
+        requirements_path=os.path.join(os.path.dirname(playbook_path), 'requirements.txt'),
+        inventory_str=inventory,
+        vars_str=(
+            'local_output_dir: {output_dir}\n'
+            'local_output_filename: user_statistics\n'
+            'remote_output_filename: /tmp/activity_report\n'
+            'config_section: {config_section}\n'
+            'extra_script_arguments: {extra_script_arguments}'
+        ).format(
+            output_dir=playbook_output_dir,
+            config_section=name_prefix,
+            extra_script_arguments=playbook_extra_script_arguments
+        ),
+        playbook_path=playbook_path,
+        username=settings.INSTANCE_LOGS_SERVER_SSH_USERNAME,
+        logger_=log_line,
+    )
+
+
+# Run on the 1st of every month
+@db_periodic_task(crontab(day='1', hour='2', minute='0'))
+def send_trial_instance_report(recipients=["patrick@opencraft.com"]):
+    """
+    Generate and send a trial instance data report
+
+    This task runs on the first of every month at 2AM
+    """
+    csvfile = StringIO()
+    csv_writer = csv.writer(csvfile, quoting=csv.QUOTE_NONNUMERIC)
+    csv_writer.writerow([
+        'Fully Qualified Domain Name',
+        'Name',
+        'Contact Email',
+        'Status',
+        'Unique Hits',
+        'Total Hits',
+        'Total Courses',
+        'Total Users',
+        'Age (Days)'
+    ])
+
+    # Get instances with an active betatestapplication and
+    # whose InstanceReference has an active OpenEdXAppServer
+    instances = OpenEdXInstance.objects \
+        .exclude(betatestapplication__isnull=True) \
+        .filter(ref_set__openedxappserver_set___is_active=True)
+
+    # Start with current date
+    today = datetime.utcnow().date()
+
+    # Get last month by subtracting the day of the month from
+    # the actual date to get the last day of last month
+    # Then replace the day with 1, to get the first day of last month
+    end_of_last_month = (today - timedelta(days=today.day))
+    beginning_of_last_month = end_of_last_month.replace(day=1)
+
+    for instance in instances:
+        betatest_application = instance.betatestapplication_set.get()
+        appserver = instance.get_active_appservers().first()
+        name_prefix = appserver.server_name_prefix
+        public_ip = appserver.server.public_ip
+
+        with ansible.create_temp_dir() as playbook_output_dir:
+            get_elasticsearch_hits_data_summary(playbook_output_dir, name_prefix, beginning_of_last_month, end_of_last_month)
+            get_instance_usage_data(playbook_output_dir, name_prefix, public_ip)
```

Comment:
Currently, this is looping through relevant instances and calling both playbooks synchronously. This should probably be streamlined/optimized to run as much in parallel as possible. The only way I can think of doing so thus far would be to gather the public IP addresses of each instance, and then run `get_instance_usage_data` with the entire list of public IPs. Even then, Elasticsearch queries would still likely need to be run sequentially.

Do you have any thoughts/suggestions?